include/bout/index_derivs_interface.hxx

```diff
@@ -207,14 +207,13 @@ template <typename T>
 T DDY(const T& f, CELL_LOC outloc = CELL_DEFAULT, const std::string& method = "DEFAULT",
       REGION region = RGN_NOBNDRY) {
   AUTO_TRACE();
-  if (std::is_base_of<Field3D, T>::value && f.hasYupYdown()
-      && ((&f.yup() != &f) || (&f.ydown() != &f))) {
+  if (std::is_base_of<Field3D, T>::value && f.hasYupYdown()) {
```

Comment:
This hasn't changed, but this condition means that Field2D always goes through the "aligned" branch. It should actually equally be able to go through the orthogonal branch without any issue now (the previous case was that `f.yup() != &f` would always be false for `Field2D` so it would always go through the second branch anyway, so the Field3D check was just a compile-time way of ensuring this). I'm not sure which is preferred/more efficient etc. but might be worth experimenting with.

Secondly I think you proposed providing alternative names rather than yup/ydown -- is it worth using the new name in the function here (i.e. `f.hasParallelSlices()` rather than `f.hasYupYdown()`), or have these renames not happened yet?

Reply:
It would be nice to be able to treat them the same!

Renames haven't happened, I was thinking of doing them in another PR in case there was any discussion on the names, but happy to just do them here (names can always be changed before the next release)